### PR TITLE
Add GSoC 2019 Project Report-LibrecoresCI(Nancy)

### PR DESCRIPTION
**Pull request by Nancy-Chauhan**
Added GSoC Report @oleg-nenashev @stffrdhrn

### CODE REVIEW

**Comment by imphil** [comment]
Thanks @Nancy-Chauhan for your blog post! It's great to read about the progress you made during the summer.

@oleg-nenashev @stffrdhrn This post needs a round of grammar and spell checking. Can you give @Nancy-Chauhan feedback on that?

**Comment by Nancy-Chauhan** [comment]
@imphil Thanks! I will make necessary changes and will ping you

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+1) Groovy Programming and developing Jenkins Shared Library
+
+2) Concepts of DSL and Closure
+
+3) Learned about various EDA Tools
+
+## Future engagements
+
+We are still aiming to involve more hardware projects to use generalised librecoresCI Pipeline library and tools .
```
Comment (oleg-nenashev) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]:
```suggestion
We would like to invite more hardware projects to use the LibreCores Pipeline library and development tools.
If you are interested to do so, please reach out to us in the [LibreCores CI Gitter Chat](https://gitter.im/librecores/librecores-ci).
```

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+author: Nancy Chauhan
+---
+
+Librecores provides a platform to share projects and ideas, in the area of free and open source digital hardware design. Librecores CI is an service to provide CI for hardware projects hosted on Librecores to improve user experience and reliability.
+
+Throughout these summers I worked on developing continuous integration for Hardware Projects like OpenRISC. OpenRISC creates free and open processor for embedded systems.
+
+I worked on GSoC Project from May to August 2019. In this project I worked on completely different technologies (EDA tools, processors, Docker, Jenkins , Groovy DSL & Jenkins Shared library) and GitHub organizations (OpenRISC, LibreCores, Fusesoc ). In this project I worked on the following goals:
+
+1) Recover OpenRISC CI pipelines and modify them by using standard LibreCores images
```
Comment (oleg-nenashev) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]: Different tense in all sentences within the list

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+7) Add TAP and Yosys code parsing & publishing steps to LibreCores Pipeline library
+
+8) Default Pipeline for FuseSoC-based OpenRISC projects
+
+## Community Bonding: Planning the solution
+
+During the community bonding period, we planned over the goals of project and its timeline. My mentors guided me for existing CI pipelines and solutions
+to proceed with the project.
+
+## Coding Phase
```
Comment (oleg-nenashev) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]:
```suggestion
## Coding Phases
```

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+The coding phase consisted of 3 months, in which I implemented the ideas I proposed in my GSoC Proposal.
+
+### First Phase
+
+In the first month, I mostly worked on recovering OpenRISC CI pipelines as it worked on Travis CI before. Now the the openrisc Continuous Integration (CI) suite is running in a librecores-ci-openrisc docker container.
+
+#### Links
+
+1) Blog: <a href = "http://nancychauhan.in/stories/2019/06/28/gsoc-week3_4/"><strong>Coding Phase 1: Week 3 & 4 </strong></a>
```
Comment (oleg-nenashev) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]: Why not Markdown?
```suggestion
1) Blog: [Coding Phase 1: Week 3 & 4](http://nancychauhan.in/stories/2019/06/28/gsoc-week3_4/)
```

**Comment by Nancy-Chauhan** [comment]
@imphil @oleg-nenashev Have done changes! Thank you for your feedback

**Comment by Nancy-Chauhan** [comment]
@imphil @oleg-nenashev I have done the changes ! Thanks for your feedback

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+## What did I learn in GSoC
+
+1) Various technologies i.e Docker, Jenkins
+
+1) Groovy Programming and developing Jenkins Shared Library
+
+2) Concepts of DSL and Closure
+
+3) Learned about various EDA Tools
```
Comment (oleg-nenashev) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]:
```suggestion
3) Various EDA Tools
```

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+1) Recover OpenRISC CI pipelines and modify them to use standard LibreCores images
+
+2) Optimise CI Pipelines with new test automation
+
+3) Develop Yosys Synthesis Metrics Parse and add to LibreCores CI base image
+
+4) Integration of _Yosys Synthesis_ for monitoring resource usages and publishing on Jenkins
+
+5) Exported OpenRISC test results to TAP and published on Jenkins
```
Comment (oleg-nenashev) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]:
```suggestion
5) Export OpenRISC test results to TAP and publish on Jenkins
```

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+I worked from May to August 2019. In this project I worked on contrasting technologies (EDA tools, processors, Docker, Jenkins, Groovy DSL & Jenkins Shared Pipeline library) and GitHub organizations (OpenRISC, LibreCores, FuseSoC ) with the following goals:
+
+1) Recover OpenRISC CI pipelines and modify them to use standard LibreCores images
+
+2) Optimise CI Pipelines with new test automation
+
+3) Develop Yosys Synthesis Metrics Parse and add to LibreCores CI base image
+
+4) Integration of _Yosys Synthesis_ for monitoring resource usages and publishing on Jenkins
```
Comment (oleg-nenashev) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]:
```suggestion
4) Add _Yosys Synthesis_  resource usage statistics parsing and publishing on Jenkins
```

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+5) Exported OpenRISC test results to TAP and published on Jenkins
+
+6) Move shared code of OpenRISC Pipelines to LibreCores Pipeline Library
+
+7) Add TAP and Yosys code parsing & publishing steps to LibreCores Pipeline library
+
+8) Default Pipeline for FuseSoC-based OpenRISC projects
```
Comment (oleg-nenashev) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]:
```suggestion
8) Implement Default Pipeline for FuseSoC-based OpenRISC projects
```

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+title: "GSoC Project Report: Continuous Integration for Hardware Projects on LibreCores CI"
+description: ""
+category:
+tags: ['GSoC', 'LibrecoresCI' ,'FOSSi-Foundation']
+author: Nancy Chauhan
+---
+
+Librecores provides a platform to share projects and ideas, in the area of free and open-source digital hardware design. Librecores CI is a service to provide CI for hardware projects hosted on Librecores to improve user experience and reliability.
+
+In these summers I worked on developing continuous integration solution for hardware Projects such as OpenRISC, a family of free and open-source processor implementations on the RISC architecture.
```
Comment (stffrdhrn) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]: Grammar:
**This summer** I worked on developing continuous integration **pipeline** for hardware **projects** such as OpenRISC, a family of free and open-source processor implementations on the RISC architecture.

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+I worked from May to August 2019. In this project I worked on contrasting technologies (EDA tools, processors, Docker, Jenkins, Groovy DSL & Jenkins Shared Pipeline library) and GitHub organizations (OpenRISC, LibreCores, FuseSoC ) with the following goals:
+
+1) Recover OpenRISC CI pipelines and modify them to use standard LibreCores images
```
Comment (stffrdhrn) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]: Should this be: **Enhance** OpenRISC CI pipelines...
I don't think there was something to "Recover"?

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+
+2) Optimise CI Pipelines with new test automation
```
Comment (stffrdhrn) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]: I think a better wording is: **Enhance** CI pipelines... here too. *Optimize* seems to intend you made them faster compared to the original travis pipeline.

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+
+3) Develop Yosys Synthesis Metrics Parse and add to LibreCores CI base image
```
Comment (stffrdhrn) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]: Grammar: Develop **a** [Yosys](http://www.clifford.at/yosys/) **synthesis metrics parser** and add **it** to **the** LibreCores CI base image

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 13]
```diff
@@ -0,0 +1,124 @@
+---
+layout: post
+title: "GSoC Project Report: Continuous Integration for Hardware Projects on LibreCores CI"
+description: ""
+category:
+tags: ['GSoC', 'LibrecoresCI' ,'FOSSi-Foundation']
+author: Nancy Chauhan
+---
+
+Librecores provides a platform to share projects and ideas, in the area of free and open-source digital hardware design. Librecores CI is a service to provide CI for hardware projects hosted on Librecores to improve user experience and reliability.
```
Comment (stffrdhrn) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]: Try to add links as you introduce new topics.

i.e.
[Librecores](https://www.librecores.org) provides a platform to share projects and ideas, in the area of free and open-source digital hardware design. [Librecores CI](https://www.librecores.org/static/librecores-ci) is a service to provide CI for hardware projects hosted on Librecores to improve user experience and reliability.

The more links the better.

**Comment by Nancy-Chauhan** [comment]
@stffrdhrn @oleg-nenashev  Thank you for your feedback. I have pushed changes. Please let me know if this looks fine

**Comment by Nancy-Chauhan** [comment]
@stffrdhrn @oleg-nenashev Can you please review it.

File: _posts/2019-08-23-gsoc_LibrecoresCI_Report.md [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+
+1. Blog: [Coding Phase 2 with Librecores CI, OpenRISC, Fusesoc & Jenkins][phase2report]
+
+2. Presentation: [GSoC/Outreachy 2019: Mid-term project demos, part 2][midtermpresentation]
+
+3. Demo Video: [GSoC/Outreachy 2019: Mid-term project demos, part 2][midtermdemo]
+
+### Third Phase
+
+In the third month, we completed Yosys Synthesis and published its result [LibreCoresCI Jenkins server][lcciopenrisc] with plot plugin which also led to the release of new version _0.5.0_ of _Librecores CI base_ docker image.
```
Comment (oleg-nenashev) [inline _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]: Link changelog?

Comment (Nancy-Chauhan) [reply _posts/2019-08-23-gsoc_LibrecoresCI_Report.md]: Added

**Comment by imphil** [comment]
Thanks for the update @Nancy-Chauhan! 

- Why did you modify  assets/people/PeterGielda.jpg?
- You can use inline links in markdown.

Is this ready to be merged? (It will go live as soon as I merge it.)

**Comment by Nancy-Chauhan** [comment]
Yes, @imphil. I have made the blog related changes. It is ready to publish

**Comment by imphil** [comment]
@Nancy-Chauhan please also have a look at my other questions, and don't touch the image of Peter.

**Comment by Nancy-Chauhan** [comment]
Yes @imphil, Just pushing necessary changes, I apologize for changing it by mistake

**Comment by Nancy-Chauhan** [comment]
I have rebased the branch. Does it look good now @imphil

**Comment by imphil** [comment]
This is now merged and will appear on the homepage and on planet librecores in the next couple of minutes.

Thanks for your work during GSoC and for keeping us updated on your success!

**Pull request merged**